apps/cli-daemon/src/app/workspace/dto/project.dto.ts

```diff
@@ -0,0 +1,17 @@
+import { Project } from '@angular-cli-gui/shared/data';
+import { JsonValue } from '@angular-devkit/core';
+import { ProjectDefinition } from '@angular-devkit/core/src/workspace';
+
+export class ProjectDto implements Project {
+  root: string;
+  prefix?: string | undefined;
+  sourceRoot?: string | undefined;
+  extensions: Record<string, JsonValue | undefined>;
+
+  constructor(projectDefinition: ProjectDefinition) {
```

Comment:
can't you use "private root: string; private prefix?: string" in the construct signature instead of duplicating the logic? (this.x = x)

Reply:
the point of this class is mapping, so how can I use `private root: string; private prefix?: string`?

Reply:
but you don't map anything you just assign x value to this.x so instead of that you can just use the "private" modifier and when you will do `new ProjectDto` you will have to pass them one-by-one and not as an object

Reply:
so, instead of doing it once in the class constructor, you offer me to do it in every place in the code I create the instance?